### PR TITLE
[WIP] Implement consumer group action items

### PR DESCRIPTION
**Pull request by hemahg**
**What this PR does / why we need it:**

- [ ] Integrate consumer group delete modal 
- [ ] Integrate consumer group view partition Drawer
- [ ] Integrate consumer group reset offset modal

### CODE REVIEW

File: packages/consoledot-api/src/mutations/useDeleteConsumerMutation.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      }
+    },
+    {
+      onSuccess: (_, { adminUrl, instanceId }) => {
+        void queryClient.invalidateQueries([
+          kafkaQueries.instance._root({
+            id: instanceId,
+            adminUrl,
+          }),
+        ]);
```
Comment (riccardo-forina) [inline packages/consoledot-api/src/mutations/useDeleteConsumerMutation.ts:39]: There is really no need to invalidate the whole cache for this. We could invalidate the consumer groups related one tho.

File: apps/consoledot-rhosak/src/routes/data-plane/routes/ConsumerGroupsRoute.tsx [hunk 2]
```diff
@@ -46,6 +50,15 @@ export const ConsumerGroupsRoute: VoidFunctionComponent<
     direction: sortDirection,
     groupId: consumerName.chips[0],
   });
+
+  const onDelete = useCallback(
+    ({ id }) => {
+      history.push(`${ControlPlaneRouteRoot}/${instance.id}/details/consumer-groups/${id}/delete`);
```
Comment (riccardo-forina) [inline apps/consoledot-rhosak/src/routes/data-plane/routes/ConsumerGroupsRoute.tsx:56]: We should have a function to generate this string, passed to the component.

File: apps/consoledot-rhosak/src/routes/data-plane/useConsumerGate.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { useConsumer } from "consoledot-api";
+import { useRouteMatch } from "react-router-dom";
+import { DataPlaneConsumerRouteParams, DataPlaneConsumerRoutePath } from "./routesConsts";
+import { useDataPlaneGate } from "./useDataPlaneGate";
+
+export function useConsumerGate() {
```
Comment (riccardo-forina) [inline apps/consoledot-rhosak/src/routes/data-plane/useConsumerGate.tsx:6]:
```suggestion
export function useConsumerGroupGate() {
```

File: apps/consoledot-rhosak/src/routes/data-plane/useConsumerGate.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  const { instance } = useDataPlaneGate();
+
+  const match = useRouteMatch<DataPlaneConsumerRouteParams>(
+    DataPlaneConsumerRoutePath
+  );
+
+  if (!match) {
+    throw Error("useDataPlaneGate used outside the expected route");
```
Comment (riccardo-forina) [inline apps/consoledot-rhosak/src/routes/data-plane/useConsumerGate.tsx:14]:
```suggestion
    throw Error("useConsumerGroup used outside the expected route");
```

File: packages/consoledot-api/src/queries/useConsumerGroup.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+  return useQuery({
+    queryKey: kafkaQueries.instance.consumer(params),
+    queryFn: () => {
+      if (!params.id) {
+        return Promise.reject("Invalid id");
+      }
+      if (!params.adminUrl) {
+        return Promise.reject("Invalid adminUrl");
+      }
```
Comment (riccardo-forina) [inline packages/consoledot-api/src/queries/useConsumerGroup.ts:26]:
```suggestion
      if (!params.adminUrl) {
        return Promise.reject("Invalid adminUrl");
      }
      if (!params.consumerId) {
        return Promise.reject("Invalid consumerId");
      }
```

File: packages/consoledot-api/src/queryKeys.ts [hunk 6]
```diff
@@ -108,6 +109,21 @@ export const kafkaQueries = {
         },
         params,
       ] as const,
+      consumer: ({
```
Comment (riccardo-forina) [inline packages/consoledot-api/src/queryKeys.ts:112]:
```suggestion
      consumerGroup: ({
```

File: packages/consoledot-api/src/queries/useConsumerGroup.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  params: { id?: string; adminUrl?: string } & Omit<
+    FetchConsumerParams,
+    "getConsumerGroupById"
+  >
+) {
+  const { refetchInterval } = useApiConfiguration();
+  const { consumerGroups } = useApi();
+
+  return useQuery({
+    queryKey: kafkaQueries.instance.consumer(params),
```
Comment (riccardo-forina) [inline packages/consoledot-api/src/queries/useConsumerGroup.ts:19]:
```suggestion
    queryKey: kafkaQueries.instance.consumerGroup(params),
```

File: packages/consoledot-api/src/transformers/consumerTransformer.ts [hunk 8]
```diff
@@ -0,0 +1,25 @@
+import { ConsumerGroupState } from '@rhoas/kafka-instance-sdk';
+import type { ConsumerGroup as ConsumerGroupApi } from "@rhoas/kafka-instance-sdk/dist/generated/model/consumer-group";
+import type { ConsumerGroup} from "ui-models/src/models/consumer-group";
+
+export function consumerTransformer(t: ConsumerGroupApi): ConsumerGroup {
+      const state = stateMapping[t.state as ConsumerGroupState]
+      return {
+        groupId: t.groupId || "",
```
Comment (riccardo-forina) [inline packages/consoledot-api/src/transformers/consumerTransformer.ts:8]: Don't default to empty string. Throw an error if the data we require to be there isn't there.

File: packages/consoledot-api/src/transformers/consumerTransformer.ts [hunk 9]
```diff
@@ -0,0 +1,25 @@
+import { ConsumerGroupState } from '@rhoas/kafka-instance-sdk';
+import type { ConsumerGroup as ConsumerGroupApi } from "@rhoas/kafka-instance-sdk/dist/generated/model/consumer-group";
+import type { ConsumerGroup} from "ui-models/src/models/consumer-group";
+
+export function consumerTransformer(t: ConsumerGroupApi): ConsumerGroup {
```
Comment (riccardo-forina) [inline packages/consoledot-api/src/transformers/consumerTransformer.ts:5]:
```suggestion
export function consumerGroupTransformer(t: ConsumerGroupApi): ConsumerGroup {
```

**Review by riccardo-forina — CHANGES_REQUESTED** [review]
The changes are fine, but please:

* re-read the code when you copy-paste it
* use meaningful names for functions and variables. A consumer is something that consumes the data from a topic, a consumer group is a label to describe consumers. Different things.

**Comment by riccardo-forina** [comment]
Did you close this by mistake? The changes were ok as a whole

**Comment by hemahg** [comment]
I was resolving the merge conflicts not sure how it got closed

**Pull request closed without merging**